event/datadeal/data_delivery_event.go

```diff
@@ -0,0 +1,231 @@
+	"github.com/medibloc/panacea-doracle/panacea"
+	log "github.com/sirupsen/logrus"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+)
+
+type DataDeliveryVoteEvent struct {
+	reactor event.Reactor
+}
+
+func NewDataDeliveryVoteEvent(s event.Reactor) DataDeliveryVoteEvent {
+	return DataDeliveryVoteEvent{s}
+}
+
+func (e DataDeliveryVoteEvent) GetEventType() string {
+	return "data_delivery"
+}
+
+func (e DataDeliveryVoteEvent) GetEventAttributeKey() string {
+	return "vote_status"
+}
+
+func (e DataDeliveryVoteEvent) GetEventAttributeValue() string {
+	return "'started'"
+}
+
+func (e DataDeliveryVoteEvent) EventHandler(event ctypes.ResultEvent) error {
+
+	dealIDStr := event.Events[datadealtypes.EventTypeDataDeliveryVote+"."+datadealtypes.AttributeKeyDealID][0]
+	dataHash := event.Events[datadealtypes.EventTypeDataDeliveryVote+"."+datadealtypes.AttributeKeyDataHash][0]
+
+	dealID, err := strconv.ParseUint(dealIDStr, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	voteOption, deliveredCid, err := e.verifyAndGetVoteOption(dealID, dataHash)
+	if err != nil {
+		if voteOption == oracletypes.VOTE_OPTION_UNSPECIFIED {
+			return fmt.Errorf("can't vote due to error while verify. dealID(%d). dataHash(%s): %v", dealID, dataHash, err)
+		} else {
+			log.Infof("vote NO due to error while verify. dealID(%d). dataHash(%s): %v", dealID, dataHash, err)
+		}
+	}
+
+	msgVoteDataDelivery, err := e.makeDataDeliveryVote(
+		e.reactor.OracleAcc().GetAddress(),
+		dataHash,
+		deliveredCid,
+		dealID,
+		voteOption,
+		e.reactor.OraclePrivKey().Serialize(),
+	)
+	if err != nil {
+		return fmt.Errorf("make DataDeliveryVote failed. dealID(%d). dataHash(%s): %v", dealID, dataHash, err)
+	}
+
+	txBuilder := panacea.NewTxBuilder(*e.reactor.QueryClient())
+
+	txBytes, err := e.generateTxBytes(msgVoteDataDelivery, e.reactor.OracleAcc().GetPrivKey(), e.reactor.Config(), txBuilder)
+	if err != nil {
+		return fmt.Errorf("generate tx failed. dealID(%d). dataHash(%s): %v", dealID, dataHash, err)
+	}
+
+	if err := e.broadcastTx(e.reactor.GRPCClient(), txBytes); err != nil {
+		return fmt.Errorf("broadcast transaction failed. dealID(%d). dataHash(%s): %v", dealID, dataHash, err)
+	}
+
+	return nil
+
+}
+
+func (e DataDeliveryVoteEvent) verifyAndGetVoteOption(dealID uint64, dataHash string) (oracletypes.VoteOption, string, error) {
+
+	dataSale, err := e.reactor.QueryClient().GetDataSale(dealID, dataHash)
+	if err != nil {
+		return oracletypes.VOTE_OPTION_UNSPECIFIED, "", err
+	}
+
+	if dataSale.Status != datadealtypes.DATA_SALE_STATUS_DELIVERY_VOTING_PERIOD {
+		return oracletypes.VOTE_OPTION_UNSPECIFIED, "", errors.New("datasale status is not DATA_SALE_STATUS_DELIVERY_VOTING_PERIOD")
+	}
+
+	if len(dataSale.VerifiableCid) == 0 {
+		return oracletypes.VOTE_OPTION_NO, "", errors.New("there is no verifiableCid")
+	}
+
+	deal, err := e.reactor.QueryClient().GetDeal(dataSale.DealId)
+	if err != nil {
+		if errors.Is(err, datadealtypes.ErrDealNotFound) {
```

Comment:
I think if does not exist deal, it does not return  `datadealtypes.ErrDealNotFound` error.
Maybe return to `"empty value"`. 
So check my code please.(https://github.com/medibloc/panacea-doracle/pull/91/files#diff-0c6d9bfa1427c39a9489d371dd13d4d5f847f7a562817ee976de39cefadf3bc1R110)

Reply:
Thanks, I understand it. Fixed in [5246329](https://github.com/medibloc/panacea-doracle/pull/93/commits/524632931d95a3fd57d53d3ed561408287e664f1).